src/Cake.NuGet/NuGetPackageInstaller.cs

```diff
@@ -149,7 +126,16 @@ public IReadOnlyCollection<IFile> Install(PackageReference package, PackageType
             }
 
             // Install the package.
-            InstallPackage(package, path);
+            if (!InstallPackage(package, path))
+            {
+                _log.Warning("An error occured while installing package {0}.", package.Package);
+                if (createdDirectory)
+                {
+                    _log.Debug("Deleting package directory {0}...", path);
+                    root.Delete(true);
+                    return new List<IFile>(0);
```

Comment:
We're not extremely concerned about allocations. Especially since this is not a hot path.